src/sql/workbench/browser/modelComponents/table.component.ts

```diff
@@ -489,4 +489,16 @@ export default class TableComponent extends ComponentBase<azdata.TableComponentP
 	public get headerFilter(): boolean {
 		return this.getPropertyOrDefault<boolean>((props) => props.headerFilter, false);
 	}
+
+	public doAction(action: string, ...args: any[]): void {
+		switch (action) {
+			case ModelViewAction.AppendData:
+				this.appendData(args[0]);
+		}
+	}
+	private appendData(data: any[][]) {
```

Comment:
Will the new data be sorted/filtered correctly as well when the layout is redone?

Reply:
Yes. It will use existing filter values.